smarkets/streaming_api/client.py

```diff
@@ -48,19 +48,43 @@ def __init__(self, session):
         self.callbacks = dict((callback_name, Signal())
                               for callback_name in self.__class__.CALLBACKS)
         self.global_callback = Signal()
+        self.last_login = None
 
     def login(self, receive=True):
         "Connect and ensure the session is active"
         self.session.connect()
         if receive:
             self.read()
+            self.check_login()
             self.flush()
 
+    def check_login(self):
+        """
+        Check if login was successful.
+        Must be executed after a 'read()' and throws an exception if not successful
+        """
+        if not self.last_login:
+            raise LoginError(eto.LOGOUT_NONE)
```

Comment:
@sandrovicente I'm wondering if we should raise a `LoginTimeOut` or something here instead. The rationale is that users catching this exception may be confused thinking that a `logout` message was received when the actual issue would be that no such message was received.

Reply:
Ok. Changed to throw 'LoginTimeout'